pyquil/operator_estimation.py

```diff
@@ -768,27 +798,48 @@ def measure_observables(qc: QuantumComputer, tomo_experiment: TomographyExperime
         for qubit, op_str in max_weight_out_op:
             total_prog += _local_pauli_eig_meas(op_str, qubit)
 
-        # 2. Run the experiment
-        bitstrings = qc.run_and_measure(total_prog, n_shots)
+        if symmetrize:
+            # 1.4 Symmetrize -- flip qubits pre-measurement
+            qubits = max_weight_out_op.get_qubits()
+            ops_strings = _ops_strs_symmetrize(qubits)
+            list_bitstrings_symm = []
+            for ops_str in ops_strings:
+                total_prog_symm = total_prog.copy()
+                prog_symm = _ops_str_to_prog(ops_str, qubits)
+                total_prog_symm += prog_symm
+                # 2. Run the experiment
+                n_shots_symm = n_shots // len(ops_strings)
+                bitstrings_symm = qc.run_and_measure(total_prog_symm, n_shots_symm)
+                # 2.1 Flip the results post-measurement
+                d_flips_symm = _ops_str_to_flips(ops_str, qubits)
+                for qubit, bs_results in bitstrings_symm.items():
+                    bitstrings_symm[qubit] = bs_results ^ d_flips_symm.get(qubit, 0)
+                # 2.2 Gather together the symmetrized results into list
+                list_bitstrings_symm.append(bitstrings_symm)
+
+            # 2.3 Gather together all the symmetrized results
+            bitstrings = reduce(lambda d1, d2: _stack_dicts(d1, d2), list_bitstrings_symm)
```

Comment:
this `lambda` is redundant. pass the function as the first argument to reduce

Reply:
functional 4 lyfe

Reply:
Updated.